bam_masterdata/excel/excel_to_entities.py

```diff
@@ -12,125 +14,56 @@
-            },
-        },
-        "EXPERIMENT_TYPE": {
-            "Code": {"key": "code", "pattern": r"^\$?[A-Za-z0-9_.]+$"},
-            "Description": {
-                "key": "description",
-                "pattern": r".*",
-                "is_description": True,
-            },
-            "Validation script": {
-                "key": "validationPlugin",
-                "pattern": r"^[A-Za-z0-9_]+\.py$",
-                "allow_empty": True,
-            },
-        },
-        "DATASET_TYPE": {  # This was missing
-            "Code": {"key": "code", "pattern": r"^\$?[A-Za-z0-9_.]+$"},
-            "Description": {
-                "key": "description",
-                "pattern": r".*",
-                "is_description": True,
-            },
-            "Validation script": {
-                "key": "validationPlugin",
-                "pattern": r"^[A-Za-z0-9_]+\.py$",
-                "allow_empty": True,
-            },
-        },
-        "PROPERTY_TYPE": {
-            "Code": {"key": "code", "pattern": r"^\$?[A-Za-z0-9_.]+$"},
-            "Description": {
-                "key": "description",
-                "pattern": r".*",
-                "is_description": True,
-            },
-            "Property label": {"key": "label", "pattern": r".*"},
-            "Data type": {
-                "key": "dataType",
-                "pattern": None,
-                "is_data": True,  # Explicitly enforce allowed types
-            },
-            "Vocabulary code": {
-                "key": "vocabularyCode",
-                "pattern": r"^\$?[A-Za-z0-9_.]+$",
-            },
-            "Metadata": {"key": "metadata", "pattern": r".*"},
-            "Dynamic script": {
-                "key": "plugin",
-                "pattern": r"^[A-Za-z0-9_]+\.py$",
-                "allow_empty": True,
-            },
-        },
-        "VOCABULARY_TYPE": {  # This was missing
-            "Code": {"key": "code", "pattern": r"^\$?[A-Za-z0-9_.]+$"},
-            "Description": {
-                "key": "description",
-                "pattern": r".*",
-                "is_description": True,
-            },
-            "Url template": {
-                "key": "url_template",
-                "pattern": r"https?://(?:www\.)?[a-zA-Z0-9-._~:/?#@!$&'()*+,;=%]+",
-                "is_url": True,
-                "allow_empty": True,  # Allows empty but validates URLs if present
-            },
-        },
-    }
+    VALIDATION_RULES: dict[str, dict[str, dict[str, Any]]] = {}
 
     def __init__(self, excel_path: str):
         """Initialize the MasterdataExtractor."""
         self.excel_path = excel_path
         self.workbook = openpyxl.load_workbook(excel_path)
         self.logger = logger
 
+        # Load validation rules at initialization
+        if not MasterdataExcelExtractor.VALIDATION_RULES:
+            self.load_validation_rules()
+
+    @classmethod
+    def load_validation_rules(cls, file_path: str = None):

```

Comment:
I think this function is not only for `MasterdataExcelExtractor`. I will take it out, probably to utils or to a separate folder of the CLI (maybe we want something like `bam_masterdata/validation_rules/`), and put this function there.

Then you just need to change the lines where `self.VALIDATION_RULES` is called.